### PR TITLE
parse & unparse on :multi

### PR DESCRIPTION
**Pull request by ikitommi**
```clj
(def Multi
  [:multi {:dispatch :type}
   [:user [:map [:size :int]]]
   [::m/default :any]])

(m/parse Multi {:type :user, :size 1})
; => [:user {:type :user, :size 1}]

(m/parse Multi {:type "sized", :size 1})
; => [:malli.core/default {:type "sized", :size 1}]

(->> {:type :user, :size 1}
     (m/parse Multi)
     (m/unparse Multi))
; => {:type :user, :size 1}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1301,6 +1301,21 @@ Schemas can be used to parse values using `m/parse` and `m/parser`:
 ;               :children [[:primitive [:text "Hello, world of data"]]]}]]}]
 ```
 
+Parsing returns tagged values for `:orn`, `:catn`, `:altn` and `:multi`. For others it returns the value.
```
Comment (nilern) [inline README.md:1304]: The value is also not returned unchanged if there are e.g. nested `:orn`:s: `[:tuple [:orn [:foo ...` => `[(miu/-tagged :foo ...`

File: src/malli/core.cljc [hunk 2]
```diff
@@ -1110,19 +1107,14 @@
                    (explainer x in acc)
                    (conj acc (miu/-error (->path path) (->path in) this x ::invalid-dispatch-value))))))
            (-parser [_]
-             (let [find (finder (reduce-kv (fn [acc k s] (assoc acc k (-parser s))) {} dispatch-map))]
-               (fn [x]
-                 (if-some [parser (find (dispatch x))]
-                   (parser x)
-                   ::invalid))))
+             (let [parse (fn [k s] (as-> (-parser s) $ (fn [x] (miu/-map-valid #(miu/-tagged k %) ($ x)))))
```
Comment (nilern) [inline src/malli/core.cljc:1110]: `as->` has no advantage over `let` here.

File: src/malli/core.cljc [hunk 3]
```diff
@@ -1110,19 +1107,14 @@
...
-             (let [unparsers (mapv (fn [[_ _ s]] (-unparser s)) children)]
-               (fn [x]
-                 ;; Can't use `dispatch` as `x` might not be valid before it has been unparsed:
-                 (reduce (fn [_ unparser] (miu/-map-valid reduced (unparser x)))
-                         ::invalid unparsers))))
+             (let [unparsers (reduce-kv (fn [acc k s] (assoc acc k (-unparser s))) {} dispatch-map)]
+               (fn [x] (if (miu/-tagged? x) (if-some [f (unparsers (key x))] (f (val x)) ::invalid) ::invalid))))
            (-transformer [this transformer method options]
-             ;; FIXME: Probably should not use `dispatch`, see comment in `-unparser` above.
+             ;; FIXME: Probably should not use `dispatch`
```
Comment (nilern) [inline src/malli/core.cljc:1117]: Add the deleted comment here instead of just removing the reference as well.

File: test/malli/core_test.cljc [hunk 4]
```diff
@@ -944,14 +944,14 @@
...
+      (is (= (miu/-tagged :sized valid1) (m/parse schema valid1)))
+      (is (= (miu/-tagged :human valid2) (m/parse schema valid2)))
       (is (= ::m/invalid (m/parse schema invalid1)))
       (is (= ::m/invalid (m/parse schema invalid2)))
       (is (= ::m/invalid (m/parse schema invalid3)))
       (is (= ::m/invalid (m/parse schema "not-a-map")))
-      (is (= valid1 (m/unparse schema valid1)))
-      (is (= valid2 (m/unparse schema valid2)))
+      (is (= valid1 (m/unparse schema (m/parse schema valid1))))
+      (is (= valid2 (m/unparse schema (m/parse schema valid2))))
```
Comment (nilern) [inline test/malli/core_test.cljc:954]: I would prefer just `(m/unparse schema (miu/-tagged :sized valid1))`, I get a bad vibe from using the SUT when creating the test inputs (even if it is a different part, tested separately).

**Review by nilern — CHANGES_REQUESTED** [review]
The README needs to be fixed at least.

**Comment by nilern** [comment]
Let's not forget that this fixes #402

File: README.md (hunk 1 above)
Comment (ikitommi) [reply README.md:1304]: good catch.

**Comment by ikitommi** [comment]
thanks for the review! fixed all but the test.

**Pull request merged**